widgets/tools/skeletonview.cpp

```diff
@@ -638,17 +639,18 @@ SkeletonView::SkeletonView(QWidget * const parent) : QWidget{parent}
         treeContextMenu.actions().at(i++)->setEnabled(selectedTrees.size() > 0);//hide
         treeContextMenu.actions().at(i++)->setEnabled(selectedTrees.size() > 0);//restore default color
         treeContextMenu.actions().at(i++)->setVisible(selectedTrees.size() > 0 && containsMesh);//remove mesh
-        treeContextMenu.actions().at(deleteActionIndex = i++)->setEnabled(selectedTrees.size() > 0);//delete
+        treeContextMenu.actions().at(deleteActionIndex = i++)->setEnabled(nodeEditing && selectedTrees.size() > 0);//delete
         //display the context menu at pos in screen coordinates instead of widget coordinates of the content of the currently focused table
         treeContextMenu.exec(treeView.viewport()->mapToGlobal(pos));
         // make some actions always available when ctx menu isn’t shown
         treeContextMenu.actions().at(copyActionIndex)->setEnabled(true);
-        treeContextMenu.actions().at(deleteActionIndex)->setEnabled(true);
+        treeContextMenu.actions().at(deleteActionIndex)->setEnabled(nodeEditing);
```

Comment:
This is not right. The reason those are always enabled is that these are shortcuts and there is nothing set up for when the conditions, which specify when they are available (e.g. changed work mode), change. The logic specifying these conditions is therefore in the actions callback.